src/middleware/types.ts

```diff
@@ -0,0 +1,16 @@
+export type GeneralRequest = {
+  method: string;
+  url: string;
+  headers: Record<string, string>;
+  text: () => Promise<string>;
+};
+
+export type GeneralResponse = {
+  status: number;
+  headers?: Record<string, string>;
+  text?: string;
```

Comment:
can we rename `text` to `body`? That would fit the convention we have in https://github.com/octokit/endpoint.js which also has the concept of "generic request options":
<img width="864" alt="image" src="https://user-images.githubusercontent.com/39992/125670471-22c45777-aad5-4e05-9bc0-08a460d91aaa.png">

See also [`RequestOptions` type in `@octokit/types](https://github.com/octokit/types.ts/blob/579aa75c8cf8c90997cc52f0f3145da8bad232d6/src/RequestOptions.ts)

Reply:
I'm conflicted about text vs body, property vs getter vs function ... It's just internal so it's easy enough to change later I guess

Reply:
When I see `bodyString` somewhere (even locally declared) then my OCD starts enslaving me to use  `bodyJSON` wherever the body is `JSON` parsed (for request) or to be serialized (for response).

I’d like to stick to a common standard (even a flawed one as long as I’m not the one to blame).

- Node.js is not a good reference since it exposes an evented API (`data/end/etc.`).
- Express defines an untyped `.body` property on the request, the concrete type relies on the body parser (or `undefined` if there’s no one). I guess that’s the (partial) reason it is named `body`.
- [Response in Fetch API](https://developer.mozilla.org/en-US/docs/Web/API/Response/Response) names its first parameter `body` which accepts a string among `FormData/ReadableStream/etc.` but does not accept an object. I think this explicitness is good. The Request object is explicit in exposing a `text()` async function.

This is why I explicitly named the property `text` in both `OctokitRequest` and `OctokitResponse`. I do respect the maintainers’ final decision: if `text` is inappropriate, just let me know.

Reply:
let's leave it as is for now, thanks 👍🏼